spacy/ml/models/textcat.py

```diff
@@ -59,7 +59,7 @@ def build_simple_cnn_text_classifier(
             resizable_layer=resizable_layer,
         )
     model.set_ref("tok2vec", tok2vec)
-    model.set_dim("nO", nO)  # type: ignore  # TODO: remove type ignore once Thinc has been updated
+    model.set_dim("nO", cast(int, nO))
```

Comment:
Could you explain this cast? `nO` could still be `None` right? If so, I'd expect this to be:

```suggestion
    if nO is not None:
        model.set_dim("nO", cast(int, nO))
```

Also, I am not sure I understand why this `set_dim` is necessary. Does the chaining fail to set the value incorrectly when `nO` is provided?

Reply:
Fair enough! We can leave the `set_dim`s here.